lib/ansible/modules/identity/keycloak/keycloak_scope.py

```diff
@@ -0,0 +1,472 @@
+                    - Boolean specifying whether this role is only granted when a scope parameter
+                      with this role name is used during the auth/token request.
+
+    clientroles:
+        description:
+            - A dict of lists of role representations (in dict-form) to be acted on. At minimum,
+              C(name) is required in a role representation. The dict's keys are the client-role
+              scopes for the attached roles representations. There is an example below. Either
+              C(roles) or C(clientroles) is required, depending on C(type). More information on
+              valid role representations is provided in the documentation for I(roles). Roles need
+              to exist already to be put in scopes.
+
+
+extends_documentation_fragment:
+    - keycloak
+
+author:
+    - Eike Frost (@eikef)
+'''
+
+EXAMPLES = '''
+- name: Add realm roles to a client scope mapping (using implicit default of acting on a client and setting realm scope mappings)
+  local_action:
+    module: keycloak_scope
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    realm: testrealm
+    state: present
+    client_id: testclient
+    roles:
+      - name: testrole01
+      - name: testrole02
+      - name: testrole03
+
+- name: Add client roles to a client-template scope mapping
+  local_action:
+    module: keycloak_scope
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    realm: testrealm
+    target: client-template
+    state: present
+    type: client
+    name: testclienttemplate
+    clientroles:
+      testclient01:
+        - name: testrole01
+      testclient02:
+        - name: testrole02
+
+- name: Remove client role testrole01 for testclient01 from testclienttemplate
+  local_action:
+    module: keycloak_scope
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    realm: testrealm
+    target: client-template
+    state: absent
+    type: client
+    name: testclienttemplate
+    clientroles:
+      testclient01:
+        - name: testrole01
+
+- name: Remove all client roles from testclient01 using exclusive and an empty client role dict
+  local_action:
+    module: keycloak_scope
+    auth_client_id: admin-cli
+    auth_keycloak_url: https://auth.example.com/auth
+    auth_realm: master
+    auth_username: USERNAME
+    auth_password: PASSWORD
+    realm: testrealm
+    state: exclusive
+    type: client
+    client_id: testclient01
+    clientroles:
+'''
+
+RETURN = '''
+msg:
+  description: Message as to what action was taken
+  returned: always
+  type: string
+  sample: "Added/Updated 0 scope mapping(s), removed 0 scope mapping(s)."
+
+proposed:
```

Comment:
There is a slight, but not complete, redundancy. Diff mode only understands before and after -- proposed is the changeset in typed form (instead of as a diff -U output which knows no types), and can be useful/extra information when "before" and "after" are the result of an API call (this would make it visible that a proposed change was accepted by the API but changed or dropped somehow).

I found these fields to be useful for register: x in tasks where I do not necessarily want to see the output of a diff in the log, but do want to use the result of operations in subsequent tasks. diff is only exposed when the diff flag is set, and while you can set that on a per-task level I don't think you can suppress its output.